Inc/info.h

```diff
@@ -0,0 +1,103 @@
+/**
+@file vport.h
+@brief Generates the device info blob and writes it over serial.
+*/
+
+#pragma once
+#include "TASRun.h"
+#include <string.h>
+#include <stdlib.h>
+
+/*****Preprocessor shenanigans*****/
+#define STRINGIFY2( x) #x
```

Comment:
What exactly are STRINGIFY and STRINGIFY2 doing? Wouldn't it be safer to wrap these macros in an extra set of parens to be safe? Is that why you have them nested like that? Can you not just do #define STRINGIFY(x) (#x) ?

Reply:
STRINGIFY is a macro to take an existing defined value (for example, `#define GREAT_BAND foo_fighters`) and turn it into a string literal (for example, `printf(STRINGIFY(GREAT_BAND));` becomes `printf("foo_fighters");`). This is required because C does not typically perform macro substitution inside string literals. STRINGIFY2() is a required layer of indirection to work around the mechanics of the C preprocessor.

This is used here to take numerical values from TASRun.h and turn them into string literals.